jquery.password_field.js

```diff
@@ -64,6 +152,115 @@
             });
 
         });
-    }
 
-}(jQuery));
\ No newline at end of file
+    };
+
+    /**
+     * Adds a strength indicator under the password field.
+     * The indicator has two parts:
+     * - a strength indicator base on <a href="https://github.com/dropbox/zxcvbn">zxcvbn</a> library.
+     * - a validator based on password rules: > 8 chars, upper and lowercase chars, at lease on digit
+     *
+     * In the options the displayed texts can be controlled.
+     *
+     * Example:
+     * <code><pre>
+     * {
+     *   strength_text: {
+     *     weak: 'Weak',
+     *     okay: 'Okay',
+     *     medium: 'Medium',
+     *     good: 'Good',
+     *     strong: 'Strong',
+     *     na: ''
+     *   },
+     *   validity: {
+     *     header: 'Your password must have',
+     *     pwd_length: '8 or more characters',
+     *     pwd_upper_and_lower: 'Upper & lowercase letters',
+     *     pwd_digits: 'At least one number'
+     *   }
+     * }
+     * </pre></code>
+     */
+    $.fn.strengthIndicator = function(options) {
+
+        var settings = $.extend({
+            strength: {
+                weak: 'Weak',
+                okay: 'Okay',
+                medium: 'Medium',
+                good: 'Good',
+                strong: 'Strong',
+                na: ''
+            },
+            validity: {
+                header: 'Your password must have',
+                pwd_length: '8 or more characters',
+                pwd_upper_and_lower: 'Upper & lowercase letters',
+                pwd_digits: 'At least one number'
+            }
+        }, options);
+
+        this.each(function () {
+            var $this = $(this);
+
+            var $fragment = $('<div><div class="password-strength"> \
+                    <ul class="password-strength-indicator"> \
+                        <li class="strong good medium okay weak"></li> \
+                        <li class="strong good medium okay"></li> \
+                        <li class="strong good medium"></li> \
+                        <li class="strong good"></li> \
+                        <li class="strong"></li> \
+                    </ul> \
+                    <div class="password-strength-text"><span></span></div> \
+                </div> \
+                <div class="password-validity-header">' + settings.validity['header'] + '</div> \
+                <ul class="password-validity"> \
+                    <li class="length">' + settings.validity['pwd_length'] + '</li> \
+                    <li class="upper-and-lower">' + settings.validity['pwd_upper_and_lower'] + '</li> \
+                    <li class="digits">' + settings.validity['pwd_digits'] + '</li> \
+                </ul></div>');
+
+            $fragment.insertAfter($this);
+
+            $this.on('input', function(event) {
+
+                var password = $this.val();
+
+                var hasDigit = /\d/.test(password);
+                var hasLower = /[a-z]/.test(password);
```

Comment:
This will not work for unicode characters, e.g. ü which is a valid lower case character. Maybe we should use something along the lines of:
```
function isLowerCase(ch) {
    return ch == ch.toLowerCase() && ch != ch.toUpperCase();
}

function isUpperCase(ch) {
    return ch == ch.toUpperCase() && ch != ch.toLowerCase();
}

hasUpper = _.any(password, isLowerCase)
hasLower = _.any(password, isUpperCase)

```